### PR TITLE
Example: Deploying to Heroku using long polling

### PR DESCRIPTION
**Pull request by CikiMomogi**
Please review for grammar error :)

### CODE REVIEW

**Comment by Tecardo1** [comment]
No grammar issues found. But you could make .ts default ;)

File: heroku-longpolling.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/* 
+1. Create file named bot.ts
+---------------------------------------------- */
+import { Bot } from 'grammy';
+
+const bot = new Bot(''); // <-- put your bot token between the ''
```
Comment (wojpawlik) [inline heroku-longpolling.ts:6]: Don't commit the token.
```suggestion
if (process.env.BOT_TOKEN == null) throw new TypeError('BOT_TOKEN missing.');
const bot = new Bot(process.env.BOT_TOKEN);
```

File: heroku-longpolling.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+Our final directory structure should look like this after compiling .ts files :
+.
+├── .git/
+├── node_modules/
+├── .gitignore 
+├── package.json
+├── package-lock.json
+├── bot.ts
+├── bot.js
```
Comment (wojpawlik) [inline heroku-longpolling.ts:48]: Using root directory for build artifacts quickly gets messy.

```suggestion
├── src/bot.ts
├── lib/bot.js
```

`lib/` and `node_modules/` should be gitignored.

File: heroku-longpolling.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+├── tsconfig.json
```
Comment (wojpawlik) [inline heroku-longpolling.ts:49]: Example `tsconfig.json` would be nice. https://www.npmjs.com/package/@tsconfig/node14?

File: heroku-longpolling.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+└── Procfile
+
+3. Final step: Deploy to heroku
```
Comment (wojpawlik) [inline heroku-longpolling.ts:52]: https://blog.heroku.com/six-strategies-deploy-to-heroku

File: heroku-longpolling.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+---------------------------------------------- */
+worker: node bot.js
+/* -------------------------------------------
+
+Our final directory structure should look like this after compiling .ts files :
+.
+├── .git/
+├── node_modules/
+├── .gitignore 
+├── package.json
```
Comment (wojpawlik) [inline heroku-longpolling.ts:45]: `package.json` should contain
```jsonc
{
	"scripts": {
		"build": "tsc"
	}
}
```

**Review by wojpawlik — CHANGES_REQUESTED** [review]
Webhook is preferable for Heroku. It makes fewer requests, and can be scaled up or put to sleep.

**Comment by CikiMomogi** [comment]
> 
> 
> Webhook is preferable for Heroku. It makes fewer requests, and can be scaled up or put to sleep.

Agree. I have a plan to make the webhook version. I also already wrote the warning about the disadvantage [here ](https://github.com/CikiMomogi/examples/blob/f31ed31fd40b0cfd76b39f7d1050d82ebf508455/heroku-longpolling.ts#L26-L28)

File: heroku-longpolling.ts (hunk 1 above)
Comment (CikiMomogi) [reply heroku-longpolling.ts]: done

**Comment by KnorpelSenf** [comment]
In order to increase the visibility and accessibility of this tutorial, I would prefer to publish your work prominently on the website rather than in the comments of an example bot. If you want to show code examples (which makes sense), you could either integrate them as code blocks, or reference the corresponding example bot with a hyperlink.

Either way, it makes sense to limit the content on the example bots repository to executable code. Note that there's a way to run example bots (see README file). The file added in this PR does not compile as-is.

If you want to provide a project template as several files, we may want to add a new directory that features a self-contained example bot setup for Heroku. This would either have a README file, or an article on the website, or both.

**Comment by CikiMomogi** [comment]
Sounds good to me. I will publish them on the [website ](https://github.com/CikiMomogi/website/blob/main/site/docs/hosting/heroku.md ) then. You can close this pull request.

**Pull request closed without merging**